[ti_abusech] make `Auth key` required for authentication (#14335)
elastic/integrations: packages/ti_abusech/_dev/build/docs/README.md

## Note:

AbuseCH now requires using an `Auth Key` (API Key) in the requests for authentication.

**If you do not authenticate:**
- You will experience API rate limiting and may lose access to new data and features.

> - Starting **June 30, 2025**, authentication will be strictly mandatory and unauthenticated requests will be denied.

Reviewer: By the time this is merged, we will be after 2025-06-30, so we don't need to include any text except for the fact that requests will be denied without authentication.

> AbuseCH requires using an `Auth Key` (API Key) in the requests for authentication.
> 
> **If you do not authenticate:**
> - You will experience API rate limiting and may lose access to new data and features.
> - Requests without authentication will be denied.
> 
> To ensure uninterrupted access, the `Auth Key` configuration option is mandatory and must be provided to avoid authentication errors and rate limiting.

Reviewer: ```suggestion
AbuseCH requires using an `Auth Key` (API Key) in the requests for authentication.
Requests without authentication will be denied by the API.
```
The rest is either repetition or contradictory; there can't be any rate limiting if the requests are denied for example.
